File: YouBooking_Front/src/app/models/hotel.ts
import {StatusHotel} from "./status-hotel";
import {Users} from "./users";
import {Room} from "./room";

export class Hotel {
  private _id!:BigInt;
  private _name!:String;
  private _address!:String;
  private _city!:String;
  private _status!:StatusHotel;
  private _rooms!:Room[];
  private _manager!:Users;

  constructor(name: String, address: String, city: String, status: StatusHotel, rooms: Room[], manager: Users) {
    this._name = name;
    this._address = address;
    this._city = city;
    this._status = status;
    this._rooms = rooms;
    this._manager = manager;
  }


  get name(): String {
    return this._name;
  }

  set name(value: String) {
    this._name = value;
  }

  get address(): String {
    return this._address;
  }

  set address(value: String) {
    this._address = value;
  }

  get city(): String {
    return this._city;
  }

  set city(value: String) {
    this._city = value;
  }

  get status(): StatusHotel {
    return this._status;
  }

  set status(value: StatusHotel) {
    this._status = value;
  }

  get rooms(): Room[] {
    return this._rooms;
  }

  set rooms(value: Room[]) {
    this._rooms = value;
  }

  get manager(): Users {
    return this._manager;
  }

  set manager(value: Users) {
    this._manager = value;
  }

  get id(): BigInt {
    return this._id;
  }

}

File: YouBooking_Back/src/main/java/com/New/YouBooking/models/Hotel.java
package com.New.YouBooking.models;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import com.New.YouBooking.models.enums.StatusHotel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class,property = "id")
public class Hotel implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(unique = true)
    private String name;

    @NotNull
    @Column(unique = true)
    private String address;

    @NotNull
    private String city;

    @Enumerated(EnumType.STRING)
    private StatusHotel status;

    @OneToMany(mappedBy = "hotel",fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    @JsonIgnoreProperties("hotel")
    private List<Room> rooms;

    @ManyToOne
    @JoinColumn(name = "manager_id")
    @JsonIgnoreProperties("hotels")
    private AppUser manager;
}
